dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeSwapDirectoriesTest.java

```diff
@@ -0,0 +1,488 @@
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+/**
+ * Test for AcmeHistoricalFile. Ensure the file is created, updated, and
+ * causes refreshing of certificates.
+ */
+@RunWith(FATRunner.class)
+public class AcmeSwapDirectoriesTest {
+
+	@Server("com.ibm.ws.security.acme.fat.simple")
+	public static LibertyServer server;
+
+	protected static ServerConfiguration ORIGINAL_CONFIG;
+
+	/*
+	 * Domains that are configured and cleared before and after the class.
+	 */
+	private static final String[] DOMAINS_ALL = { "domain1.com", "domain2.com", "domain3.com" };
+	private static final String[] DOMAINS_1 = { "domain1.com" };
+	private static final String[] DOMAINS_2 = { "domain1.com", "domain2.com", "domain3.com" };
+	private static final String[] DOMAINS_3 = { "domain1.com", "domain2.com" };
+
+	public static CAContainer caContainer;
+	public static CAContainer caContainer2;
+
+	@Rule
+	public TestName testName = new TestName();
+
+	@BeforeClass
+	public static void beforeClass() throws Exception {
+		ORIGINAL_CONFIG = server.getServerConfiguration();
+		caContainer = new PebbleContainer();
+		caContainer2 = new PebbleContainer();
+		AcmeFatUtils.checkPortOpen(caContainer.getHttpPort(), 60000);
+	}
+
+	@AfterClass
+	public static void afterClass() throws Exception {
+		if (caContainer != null) {
+			caContainer.stop();
+		}
+		if (caContainer2 != null) {
+			caContainer2.stop();
+		}
+	}
+
+	@After
+	public void afterTest() throws Exception {
+		/*
+		 * Clear the DNS records for the domain. Required since a few of the
+		 * tests setup invalid A records to test failure scenarios.
+		 */
+		AcmeFatUtils.clearDnsForDomains(caContainer, DOMAINS_ALL);
+		AcmeFatUtils.clearDnsForDomains(caContainer2, DOMAINS_ALL);
+
+		/*
+		 * Cleanup any generated ACME files.
+		 */
+		AcmeFatUtils.deleteAcmeFiles(server);
+	}
+
+	/**
+	 * This test will verify that the ACME file is created and updated
+	 * properly. When the directoryURI changes, the certificate should
+	 * be refreshed.
+	 * 
+	 * @throws Exception
+	 *             If the test failed for some reason.
+	 */
+	@Test
+	@CheckForLeakedPasswords(AcmeFatUtils.CACERTS_TRUSTSTORE_PASSWORD)
+	public void update_directoryURI() throws Exception {
+
+		/*
+		 * Configure the acmeCA-2.0 feature.
+		 */
+		AcmeFatUtils.configureAcmeCA(server, caContainer, ORIGINAL_CONFIG, false, DOMAINS_1);
+
+		try {
+
+			/*
+			 * Start the server and wait for the certificate to be installed.
+			 */
+			server.startServer();
+			AcmeFatUtils.waitForAcmeToCreateCertificate(server);
+			AcmeFatUtils.waitForSslToCreateKeystore(server);
+			AcmeFatUtils.waitForSslEndpoint(server);
+
+			/*
+			 * Verify that the server is now using a certificate signed by the
+			 * CA.
+			 */
+			Certificate[] certificates1 = AcmeFatUtils.assertAndGetServerCertificate(server, caContainer);
+			/***********************************************************************
```

Comment:
I can't figure out what whitespace you are referring to. Maybe it's confusing because line 141 looks like it's two lines? Or you want me to add white space?

Reply:
btw certificate and the comment block below. Nit picky.